src/dtx/dtx_common.c

```diff
@@ -755,9 +748,12 @@ dtx_batched_commit_register(struct ds_cont_child *cont)
 
 	head = &dss_get_module_info()->dmi_dtx_batched_list;
 	d_list_for_each_entry(dbca, head, dbca_link) {
+		if (dbca->dbca_deregistering != NULL)
+			continue;
+
 		if (uuid_compare(dbca->dbca_cont->sc_uuid,
 				 cont->sc_uuid) == 0)
-			goto out;
+			return 0;
```

Comment:
Why there will be such case? With your patch, only the first open of the container will trigger register_batched_commit(), so there should be no others registered the same container, right?

Reply:
No, each open will call register_batched_commit(), right?

Reply:
No, for a given container, only the first open will trigger register_batched_commit() for such container. The subsequent open will not trigger that unless the first one is closed (deregistered).